src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts

```diff
@@ -133,37 +145,55 @@ export class CodeEditorTutorAssessmentContainerComponent implements OnInit, OnDe
-                    this.participationCouldNotBeFetched = true;
-                    if (error.error && error.error.errorKey === 'lockedSubmissionsLimitReached') {
-                        this.lockLimitReached = true;
-                    }
-                },
-                () => {
-                    this.loadingParticipation = false;
-                },
-            );
+            this.programmingExerciseParticipationService
+                .getStudentParticipationWithLatestManualResult(participationId)
+                .pipe(
+                    tap(
+                        (participationWithResult: ProgrammingExerciseStudentParticipation) => {
+                            // Set domain to make file editor work properly
+                            this.domainService.setDomain([DomainType.PARTICIPATION, participationWithResult]);
+                            this.participation = participationWithResult;
+                            this.manualResult = this.participation.results![0];
+
+                            // Either submission from latest manual or automatic result
+                            this.submission = this.manualResult.submission as ProgrammingSubmission;
+                            this.submission.participation = this.participation;
+                            this.exercise = this.participation.exercise as ProgrammingExercise;
+                            this.hasAssessmentDueDatePassed = !!this.exercise!.assessmentDueDate && moment(this.exercise!.assessmentDueDate).isBefore(now());
+
+                            this.checkPermissions();
+                            this.handleFeedback();
+
+                            if (this.manualResult && this.manualResult.hasComplaint) {
+                                this.getComplaint();
+                            }
+                            this.createAdjustedRepositoryUrl();
+                        },
+                        (error: HttpErrorResponse) => {
+                            this.participationCouldNotBeFetched = true;
+                            if (error.error && error.error.errorKey === 'lockedSubmissionsLimitReached') {
+                                this.lockLimitReached = true;
+                            }
+                        },
+                        () => (this.loadingParticipation = false),
+                    ),
+                    switchMap(() => this.programmingExerciseService.findWithTemplateAndSolutionParticipation(this.exercise.id!)),
+                    tap((programmingExercise) => (this.templateParticipation = programmingExercise.body!.templateParticipation!)),
+                    switchMap(() => {
+                        // Get all files with content from template repository
+                        this.domainService.setDomain([DomainType.PARTICIPATION, this.templateParticipation]);
+                        const observable = this.repositoryFileService.getFilesWithContent();
```

Comment:
This could hurt a bit if the repositories are known to contain larger files (e.g. some example data), but it is not dramatic. We could in future however use the already implemented file change detection to reduce this only to files that were changed.